synapse/config/jwt_config.py

```diff
@@ -29,11 +29,14 @@ def read_config(self, config, **kwargs):
         jwt_config = config.get("jwt_config", None)
         if jwt_config:
             self.jwt_enabled = jwt_config.get("enabled", False)
-            self.jwt_secret = jwt_config["secret"]
             self.jwt_algorithm = jwt_config["algorithm"]
 
             # The issuer and audiences are optional, if provided, it is asserted
             # that the claims exist on the JWT.
+            self.jwt_secret = jwt_config.get("secret")
+            self.jwt_jwks_uri = jwt_config.get("jwks_uri")
+            self.jwt_subject_claim = jwt_config.get("subject_claim")
```

Comment:
It would be clearer to have this default to `"sub"` here instead of down in the login code!

Reply:
At first I tried to specify the default value here, maybe I made a mistake, but it didn't work for some reason.

Reply:
That's surprising! We do that in many places, could you try again? I hope it is just changing this to `jwt_config.get("subject_claim", "sub")`.